src/manifests_base.py

```diff
@@ -118,3 +119,42 @@ def update_parameters(self, parameters: Dict[str, str]) -> None:
                 parameters.pop(adjustment[:-1], None)
             else:
                 log.warning("Invalid parameter: %s in %s", adjustment, config)
+
+
+class CephToleration(Toleration):
+    @classmethod
+    def from_string(cls, toleration_str: str) -> "CephToleration":
+        """Parses a toleration string into a Toleration object.
+
+        Raises:
+            ValueError: If the string is not a valid toleration.
+        """
```

Comment:
Continuing my comment about default values, maybe we can do something like this:
```suggestion
    def from_string(cls, toleration_str: str) -> Optional["CephToleration"]:
        """Parses a toleration string into a Toleration object.

        Raises:
            ValueError: If the string is not a valid toleration.
        """
        if not toleration_str:
            return None
```

Reply:
It's not possible for from_string to be called with an empty string.  the `toleration_str` here will always be some non-empty value due to the way python's `str.split()` works.  I'll make this more clear by making `from_string` a private method.